lighthouse-core/gather/gather-runner.js

```diff
@@ -119,7 +130,16 @@ class GatherRunner {
     await driver.registerPerformanceObserver();
     await driver.dismissJavaScriptDialogs();
     await driver.registerRequestIdleCallbackWrap(options.settings);
-    if (resetStorage) await driver.clearDataForOrigin(options.requestedUrl);
+    if (resetStorage) {
+      const locations = await driver.getImportantLocationsNotCleared(options.requestedUrl);
+      if (locations.length) {
+        LighthouseRunWarnings.push(str_(
+          UIStrings.warningData,
+          {locations: locations.join(', '), locationCount: locations.length}
```

Comment:
I don't believe we can concat a list of names like this, especially in the middle of a string. We have a few options, e.g. there are only three possible values, so having three separate strings:
```js
const warningData1 = 'There may be important data in this location: {location}. Audit...';
const warningData2 = 'There may be important data in these locations: {location1}, {location2}. Audit...';
const warningData3 = 'There may be important data in these locations: {location1}, {location2}, {location3}. Audit...';
```

There's also `Intl.ListFormat`, but that's not available until Node 12 (and supposedly there may still be issues with that...I'm not sure how we communicate that will be used to the translators, for instance).

Maybe there are other options, @exterkamp?

Reply:
Why not? It is working if the parameter is a string literal. Are you saying `locations` should be a list of UI strings?

Reply:
No, the locations should be replacement values, but I mean languages don't all format lists the same way, so

"_localized content:_ Local Storage, IndexedDB, Web SQL. _localized content_"

(or whatever order it ends up in) may be a nonsense way of doing a list like that in that locale.

There has been an argument that what's being stated should still be _understandable_ by someone using that locale (especially because many technical users are used to being stuck with only english in their developer tools), but in the past we've taken the position that we should stick with English for the string until we can't figure out the "right" away to do it.

Another example is the `multi-check-audit` "failures" list: 

https://github.com/GoogleChrome/lighthouse/blob/bc6ab76a33f50403e4bfc84eb069b070ed10466f/lighthouse-core/audits/multi-check-audit.js#L56-L57

The TODO points to "i18n concat'd lists" in #7238 (maybe it's finally time to do)